src/archetype.mojo

```diff
@@ -205,3 +299,42 @@ struct Archetype[dType: DType](CollectionElementNew):
             _ = self._entities.pop()
 
         return swapped
+
+    fn add(inout self, entity: Entity) -> Int:
+        """Adds an entity to the archetype.
+
+        Args:
+            entity: The entity to add.
+
+        Returns:
+            The index of the entity in the archetype.
+        """
+        if self._size == self._capacity:
+            self.reserve()
+
+        self._entities.append(entity)
+        self._size += 1
+        return self._size - 1
+
+    fn extend(
+        inout self, count: Int, inout entity_pool: EntityPool
+    ) -> Int as start_index:
+        """Adds an entity to the archetype.
```

Comment:
You are right, the doc string is outdated. The archetype simply retrieves the entities to be created from the entity pool. This saves the work of either putting all new entities into a list handed over to the function or extending with uninitialized memory. Do you think this is bad?

Reply:
See my comment on appending/inserting entities. You would have to return the created entities, so the world is able to insert them in the right position.

Better add entities to the archetype by calling a method with each of them, individually (i.e., don't pass a list to avoid the overhead of list creation). You can pre-allocate memory in the archetype in cases where you already know that you will add more than one entity.

Reply:
I have somewhat kept my structure but think that it is good now. Please have a look at the new PR